File: app/src/main/java/android/pdf4664/notepad/utilities/DocUtils.java
package android.pdf4664.notepad.utilities;

import android.content.Context;

import java.io.File;

public class DocUtils {

    private static String[] dirContents(Context c, boolean loadTemplates) {
        File file;

        if (loadTemplates) {
            file = new File(DocUtils.templatesPath(c.getApplicationContext()));
        } else {
            file = new File(DocUtils.notesPath(c.getApplicationContext()));
        }

        File[] fs = file.listFiles();

        String[] fileContents = new String[fs.length];
        for (int i = 0; i<fs.length; i++) {
            if (fs[i].getName()=="null") {
                continue;
            }
            fileContents[i] = fs[i].getName();
        }

        return fileContents;
    }

    //Utilities for the notes folder
    public static String notesPath(Context c) {
        String path = c.getFilesDir().getAbsolutePath();
        path+= File.separator;
        path+="notes"+File.separator;
        return path;
    }

    public static void checkNotesPath(Context c) {
        File f = new File(notesPath(c.getApplicationContext()));
        if (!f.exists()) {
            f.mkdirs();
        }
    }

    public static String[] notesFileContents(Context c) {
        return dirContents(c,false);
    }

    //Utilities for the templates folder
    public static String templatesPath(Context c) {
        String path = c.getFilesDir().getAbsolutePath();
        path+= File.separator;
        path+="templates"+File.separator;
        return path;
    }

    public static void checkTemplatesPath(Context c) {
        File f = new File(templatesPath(c.getApplicationContext()));
        if (!f.exists()) {
            f.mkdirs();
        }
    }

    public static String[] templateFileContents(Context c) {
        return dirContents(c,true);
    }
}

File: app/src/main/java/org/pdf4664/notepad/utilities/DocUtils.kt
package org.pdf4664.notepad.utilities

import android.content.Context
import android.util.Log

import java.io.File

object DocUtils {

    var prefix : String = ""

    private fun dirContents(c: Context, loadTemplates: Boolean): Array<String?> {
        val file: File

        if (loadTemplates) {
            file = File(DocUtils.templatesPath(c.applicationContext))
        } else {
            file = File(DocUtils.notesPath(c.applicationContext))
        }

        val fs = file.listFiles()

        val fileContents = arrayOfNulls<String>(fs.size)
        for (i in fs.indices) {
            if (fs[i].name === "null") {
                continue
            }
            fileContents[i] = fs[i].name
        }

        return fileContents
    }

    //Utilities for the notes folder
    fun notesPath(c: Context): String {
        var path = c.filesDir.absolutePath
        path += File.separator
        path += "notes" + File.separator
        path += prefix
        Log.i("PATH_INFO",path)
        return path
    }

    fun navUp() {
        prefix = prefix.dropLast(1)

        if (!prefix.contains("/")) {
            prefix = ""
            return
        }

        val last = prefix.lastIndexOf("/")
        prefix = prefix.substring(0,last)
        prefix += File.separator
    }

    fun checkNotesPath(c: Context) {
        val f = File(notesPath(c.applicationContext))
        if (!f.exists()) {
            f.mkdirs()
        }
    }

    fun notesFileContents(c: Context): Array<String?> {
        return dirContents(c, false)
    }

    //Utilities for the templates folder
    fun templatesPath(c: Context): String {
        var path = c.filesDir.absolutePath
        path += File.separator
        path += "templates" + File.separator
        return path
    }

    fun checkTemplatesPath(c: Context) {
        val f = File(templatesPath(c.applicationContext))
        if (!f.exists()) {
            f.mkdirs()
        }
    }

    fun templateFileContents(c: Context): Array<String?> {
        return dirContents(c, true)
    }
}
